File: Java/src/Java Web - ScreenMatch/screenmatch/src/main/java/br/com/alura/screenmatch/service/ConsumoAPI.java
package br.com.alura.screenmatch.service;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class ConsumoAPI {

    final String apiKey = System.getenv("API_IMDB_KEY");

    public String obterDados(String endereco){

        endereco = endereco + "&apikey=" + apiKey;
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(endereco))
                    .build();
        HttpResponse<String> response = null;
        try {
            response = client
                    .send(request, HttpResponse.BodyHandlers.ofString());
        } catch (IOException e) {
            throw new RuntimeException(e);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }

        String json = response.body();
        return json;
    }
}

File: Kotlin/alugames/src/main/kotlin/br/com/alura/alugames/servicos/ConsumoApi.kt
package br.com.alura.alugames.servicos

import br.com.alura.alugames.modelo.Gamer
import br.com.alura.alugames.modelo.InfoGamerJson
import br.com.alura.alugames.modelo.InfoJogo
import br.com.alura.alugames.modelo.Jogo
import br.com.alura.alugames.utilitario.criaGamer
import br.com.alura.alugames.utilitario.criaJogo
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse

//https://raw.githubusercontent.com/jeniblodev/arquivosJson/main/jogos.json
class ConsumoApi {

    fun consomeDados(endereco:String):String{
        val client: HttpClient = HttpClient.newHttpClient()
        val request = HttpRequest.newBuilder()
            .uri(URI.create(endereco))
            .build()
        val response = client
            .send(request, HttpResponse.BodyHandlers.ofString())

        return response.body()
    }
    fun buscaJogo(id:String):InfoJogo {
    val endereco = "https://www.cheapshark.com/api/1.0/games?id=$id"

    val json = consomeDados(endereco)

    val gson = Gson()
    val meuInfoJogo:InfoJogo = gson.fromJson(json, InfoJogo::class.java)

    return meuInfoJogo
    }

    fun buscaJogoJson():List<Jogo>{
        val endereco = "https://raw.githubusercontent.com/jeniblodev/arquivosJson/main/jogos.json"

        val json = consomeDados(endereco)

        val jogoTipo = object : TypeToken<List<Jogo>>() {}.type
        val gson = Gson()
        val listaJogos : List<Jogo> = gson.fromJson(json, jogoTipo)

        val listaJogosConvertida = listaJogos.map{ Jogo -> Jogo.criaJogo()}
        return listaJogosConvertida
    }

    fun buscaGamers(): List<Gamer> {
        val endereco = "https://raw.githubusercontent.com/jeniblodev/arquivosJson/main/gamers.json"

        val json = consomeDados(endereco)

        val gamerTipo = object : TypeToken<List<InfoGamerJson>>() {}.type
        val gson = Gson()
        val listaGamers : List<InfoGamerJson> = gson.fromJson(json, gamerTipo)

        val listaGamersConvertida = listaGamers.map{ infoGamerJson -> infoGamerJson.criaGamer()}
        return listaGamersConvertida
    }
}